packages/nextjs/components/NetworksDropdown/NetworksDropdown.tsx

```diff
@@ -0,0 +1,336 @@
+        instanceId="network-select"
+        options={Object.values(groupedOptionsState)}
+        onChange={handleSelectChange}
+        components={{ Option: props => <CustomOption {...props} onDelete={handleDeleteCustomChain} /> }}
+        isSearchable={!isMobile}
+        className="max-w-xs relative text-sm w-44"
+        theme={theme => ({
+          ...theme,
+          colors: {
+            ...theme.colors,
+            primary25: isDarkMode ? "#401574" : "#efeaff",
+            primary50: isDarkMode ? "#551d98" : "#c1aeff",
+            primary: isDarkMode ? "#BA8DE8" : "#551d98",
+            neutral0: isDarkMode ? "#130C25" : theme.colors.neutral0,
+            neutral80: isDarkMode ? "#ffffff" : theme.colors.neutral80,
+          },
+        })}
+        styles={{
+          menuList: provided => ({ ...provided, maxHeight: 280, overflow: "auto" }),
+          control: provided => ({ ...provided, borderRadius: 12 }),
+          indicatorSeparator: provided => ({ ...provided, display: "none" }),
+          menu: provided => ({
+            ...provided,
+            border: `1px solid ${isDarkMode ? "#555555" : "#a3a3a3"}`,
+          }),
+        }}
+      />
+      <dialog
+        id="see-other-chains-modal"
+        className="modal"
+        ref={seeOtherChainsModalRef}
+        onClose={handleCloseOtherChainsModal}
+      >
+        <div className="flex flex-col modal-box justify-center px-12 h-3/4 sm:w-1/2 max-w-5xl bg-base-200">
+          <div className="flex justify-between items-center mb-6">
+            <h3 className="font-bold text-xl">All Chains</h3>
+            <div className="modal-action mt-0">
+              <button className="hover:text-error" onClick={handleCloseOtherChainsModal}>
+                <XMarkIcon className="h-6 w-6" />
+              </button>
+            </div>
+          </div>
+          <input
+            type="text"
+            placeholder="Search chains..."
+            className="input input-bordered w-full mb-4 bg-neutral"
+            value={searchTerm}
+            onChange={e => setSearchTerm(e.target.value)}
+            ref={searchInputRef}
+          />
+
+          <div className="flex flex-wrap content-start justify-center gap-4 overflow-y-auto h-5/6 p-2">
+            {modalChains.map(option => (
+              <div
+                key={`${option.label}-${option.value}`}
+                className="card shadow-md bg-base-100 cursor-pointer h-28 w-60 text-center"
+                onClick={() => handleSelectOtherChainInModal(option)}
+              >
+                <div className="card-body flex flex-col justify-center items-center p-4">
+                  <span className="text-sm font-semibold">Chain Id: {option.value}</span>
+                  <span className="text-sm">{option.label}</span>
+                </div>
+              </div>
+            ))}
+          </div>
+        </div>
+      </dialog>
+
+      <dialog
```

Comment:
Umm lets create an extract this Modal into it's own component 

Example [here](https://github.com/BuidlGuidl/grants.buidlguidl.com/blob/7230af2f0e3fa886940b2901f7127149f2f92167/packages/nextjs/app/admin/_components/GrantReview.tsx#L228) and [here](https://github.com/BuidlGuidl/grants.buidlguidl.com/blob/main/packages/nextjs/app/admin/_components/ActionModal.tsx)

Reply:
[Other chains modal](https://github.com/BuidlGuidl/abi.ninja/pull/129/commits/1e8664e861797433f273912460f26fc58962ca6a)
-> Removed the search bar focus from other chains modal 

[Add custom chain modal](https://github.com/BuidlGuidl/abi.ninja/pull/129/commits/2315268653507a48a021fa928a71fa7d61126862)